benches/bench_f16_return_nan.rs

```diff
@@ -0,0 +1,83 @@
+#![feature(stdsimd)]
+
+extern crate dev_utils;
+
+#[cfg(feature = "half")]
+use argminmax::ArgMinMax;
+use codspeed_criterion_compat::*;
+use dev_utils::{config, utils};
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+use argminmax::{SIMDArgMinMax, AVX2, AVX512, SSE};
+#[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+use argminmax::{SIMDArgMinMax, NEON};
+use argminmax::{ScalarArgMinMax, SCALAR};
+
+#[cfg(feature = "half")]
+use half::f16;
+
+#[cfg(feature = "half")]
+fn get_random_f16_array(n: usize) -> Vec<f16> {
+    let data = utils::get_random_array::<u16>(n, u16::MIN, u16::MAX);
+    let data: Vec<f16> = data.iter().map(|&x| f16::from_bits(x)).collect();
+    // Replace NaNs and Infs with 0
+    let data: Vec<f16> = data
+        .iter()
+        .map(|&x| {
+            if x.is_nan() || x.is_infinite() {
+                f16::from_bits(0)
+            } else {
+                x
+            }
+        })
+        .collect();
+    data
+}
+
+// TODO: rename _random_long_ to _nanargminmax_
```

Comment:
Will do this in a separate PR (cleaning up the benchmarks; renaming + removing unused benchmarks)